lib/bes/buildbuddy.go

```diff
@@ -34,7 +36,7 @@ type BuildBuddyClient struct {
 // specified URL. If not nil, auth cookies are discovered via BaseFlags and
 // added to every request. apiKey must be a valid BuildBuddy API key (other
 // forms of auth are not currently supported).
-func NewBuildBuddyClient(u *url.URL, bf *client.BaseFlags, apiKey string) (*BuildBuddyClient, error) {
+func NewBuildBuddyClient(u *url.URL, bf *client.BaseFlags, apiKey string, client httpDoer) (*BuildBuddyClient, error) {
```

Comment:
I think this would be better as a separate function: `func NewBuildBuddyClientForTest(mockHttp httpDoer) *BuildBuddyClient` - simpler implementation - and the fact that the cookie doesn't get attached to a pre-created client doesn't become confusing if the `client != nil` codepath is used in production code

Reply:
Made a function NewTestClient() and reverted the normal factory change